From 764580dc0e8b9c2d8eae6415c7ba4de63d7a38ec Mon Sep 17 00:00:00 2001
From: Corey Bryant <corey.bryant@canonical.com>
Date: Thu, 3 Dec 2015 14:36:01 -0500
Subject: [PATCH] Enable py3 mock.patch of RuntimeError

RuntimeError is moved to builtins in Python3.x so adjust
the @mock.patch accordingly when running in Python3.x.

Change-Id: I5e1ce3ed40add635ce25bc1d80f88dc7ad9d2c59
Closes-Bug: #1522551
Forwarded: https://review.openstack.org/#/c/253165/
---
 oslo_service/tests/test_sslutils.py | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/oslo_service/tests/test_sslutils.py b/oslo_service/tests/test_sslutils.py
index 6dd7bf4..c05481f 100644
--- a/oslo_service/tests/test_sslutils.py
+++ b/oslo_service/tests/test_sslutils.py
@@ -15,6 +15,7 @@
 import mock
 import os
 import ssl
+import sys
 
 from oslo_config import cfg
 
@@ -28,6 +29,11 @@ SSL_CERT_DIR = os.path.normpath(os.path.join(
                                 os.path.dirname(os.path.abspath(__file__)),
                                 'ssl_cert'))
 
+if sys.version_info >= (3,0):
+    runtime_error = "builtins.RuntimeError"
+else:
+    runtime_error = "exceptions.RuntimeError"
+
 
 class SslutilsTestCase(base.ServiceBaseTestCase):
     """Test cases for sslutils."""
@@ -38,7 +44,7 @@ class SslutilsTestCase(base.ServiceBaseTestCase):
         self.key_file_name = os.path.join(SSL_CERT_DIR, 'privatekey.key')
         self.ca_file_name = os.path.join(SSL_CERT_DIR, 'ca.crt')
 
-    @mock.patch("exceptions.RuntimeError")
+    @mock.patch(runtime_error)
     @mock.patch("os.path.exists")
     def test_is_enabled(self, exists_mock, runtime_error_mock):
         exists_mock.return_value = True
-- 
2.6.2

